pkg/devfile/adapters/kubernetes/storage/utils.go

```diff
@@ -188,7 +188,7 @@ func generateVolumeNameFromPVC(pvc string) (volumeName string, err error) {
 
 // HandleEphemeralStorage creates or deletes the ephemeral volume based on the preference setting
 func HandleEphemeralStorage(client kclient.ClientInterface, storageClient storage.Client, componentName string) error {
-	pref, err := preference.New()
+	pref, err := preference.NewClient()
```

Comment:
I suggested the same in https://github.com/redhat-developer/odo/pull/5296#discussion_r777858651. But it seems like it was a bit too long to read. 😃 

> Like the blog you shared suggests, we should create struct that is then passed to the below layers. Currently, below three calls to `preference.NewClient()` are making me cringe a bit since they are in layers below the CLI layer:
> 
> * `pkg/devfile/adapters/kubernetes/storage/utils.go`
> 
> * `pkg/kclient/oc_server.go`
> 
> * `pkg/kclient/pods.go`